paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/shuffle/RangeShuffle.java

```diff
@@ -486,4 +548,45 @@ public int get() {
             return list.get(RANDOM.nextInt(list.size()));
         }
     }
+
+    @VisibleForTesting
+    static <T> T[] allocateRangeBaseSize(Tuple2<T, Integer>[] elements, int rangeNum) {
+        int elementsNum = elements.length;
+        long totalSize = Arrays.stream(elements).mapToLong(t -> (long) t.f1).sum();
+        long restSize = totalSize;
+        double targetSize = (totalSize) / (double) (rangeNum);
+
+        @SuppressWarnings("unchecked")
+        T[] range = (T[]) new Object[rangeNum];
```

Comment:
why not rangeNum - 1?  If we separate a array num to 1000 part， we should set 999 point?